core/class.go

```diff
@@ -112,3 +115,80 @@ func flattenSierraEntryPoints(entryPoints []SierraEntryPoint) []*felt.Felt {
 	}
 	return result
 }
+
+func cairoClassHash(class Class) (*felt.Felt, error) {
+	switch t := class.(type) {
+	case *Cairo0Class:
+		return t.Hash(), nil
+	case *Cairo1Class:
+		return t.Hash(), nil
+	default:
+		return nil, errors.New("unknown class")
+	}
+}
+
+type ErrCantVerifyClassHash struct {
+	c           Class
+	hashFailure error
+	next        *ErrCantVerifyClassHash
+}
+
+func (e *ErrCantVerifyClassHash) Unwrap() error {
+	if e != nil && e.next != nil {
+		return e.next
+	}
+	return nil
+}
+
+func (e *ErrCantVerifyClassHash) Error() string {
+	var errStr string
+	if e != nil && e.hashFailure != nil {
+		errStr = fmt.Sprintf("can't verify class hash -> %s", e.hashFailure)
+	}
+	return errStr
+}
+
+func (e *ErrCantVerifyClassHash) Class() Class {
+	return e.c
+}
+
+func verifyClassHash(c Class, hash *felt.Felt) *ErrCantVerifyClassHash {
+	if c == nil {
+		return &ErrCantVerifyClassHash{
+			hashFailure: fmt.Errorf("class is nil"),
+		}
+	}
+
+	classHash, err := cairoClassHash(c)
+	if err != nil {
+		return &ErrCantVerifyClassHash{
+			c:           c,
+			hashFailure: err,
+		}
+	}
+
+	if !classHash.Equal(hash) {
+		return &ErrCantVerifyClassHash{
+			c:           c,
+			hashFailure: fmt.Errorf("expected %s, got %s", hash.String(), classHash.String()),
+		}
+	}
+
+	return nil
+}
+
+func VerifyClassHashes(classes map[felt.Felt]Class) error {
+	var head *ErrCantVerifyClassHash
+	for hash, class := range classes {
+		if err := verifyClassHash(class, &hash); err != nil {
+			err.next = head
+			head = err
+		}
+	}
+
+	if head != nil {
+		return head
```

Comment:
```suggestion
		return *head
```

`error` accepts both `*ErrCantVerifyClassHash` and `ErrCantVerifyClassHash`, however `Is` and `As` matching for some reason frails if it is not `ErrCantVerifyClassHash`

Reply:
Fixed this